spring-integration-kafka/src/main/java/org/springframework/integration/kafka/inbound/KafkaMessageDrivenChannelAdapter.java

```diff
@@ -557,33 +536,10 @@ public void onMessage(List<ConsumerRecord<K, V>> records, Acknowledgment acknowl
 				message = toMessage(records, acknowledgment, consumer);
 			}
 			if (message != null) {
-				RetryTemplate template = KafkaMessageDrivenChannelAdapter.this.retryTemplate;
-				if (template != null) {
-					doWIthRetry(records, acknowledgment, consumer, message, template);
```

Comment:
I don't think it is too bad to have a retry for the whole batch. And I believe that's why that `KafkaInboundEndpoint` API is so generic.
Any explanations why would you like to get rid of this?
Indeed we still can use a built-in retry feature from Spring for Apache Kafka: https://docs.spring.io/spring-kafka/reference/kafka/annotation-error-handling.html#default-eh.

But then we just don't set a `RetryTemplate` on this channel adapter.

Reply:
`KafkaMessageDrivenChannelAdapter.retryTemplate` cannot be set in batch mode. Therefore, template != null will never be true.
```
	public void setRetryTemplate(RetryTemplate retryTemplate) {
		Assert.isTrue(retryTemplate == null || this.mode.equals(ListenerMode.record),
				"Retry is not supported with mode=batch");
		this.retryTemplate = retryTemplate;
	}
```
Is there another way to set a RetryTemplate for the channel adapter? Please correct me if I'm wrong.